### PR TITLE
Ability to interpolate multiple properties

### PR DESCRIPTION
**Pull request by sac-bsa**
*Describe what you did in this PR and why you did it.*
#3371 showed that the bilinear particle interpolator would fail an assertion if a particle was added during the computation since it would be called to interpolate all properties to the new particle. This PR aims to fix that. 
### For all pull requests:

* [X] I have followed the [instructions for indenting my code](../blob/master/CONTRIBUTING.md#making-aspect-better).

### For new features/models or changes of existing features:
* [X] I have tested my new feature locally to ensure it is correct.
* [X] I have [created a testcase](http://www.math.clemson.edu/~heister/manual.pdf#sec%3Awriting_tests) for the new feature/benchmark in the [tests/](../blob/master/tests/) directory.

### CODE REVIEW

File: source/particle/interpolator/bilinear_least_squares.cc [hunk 1]
```diff
@@ -84,91 +80,106 @@ namespace aspect
...
-            A(index,0) = 1;
-            A(index, 1) = relative_particle_position[0];
-            A(index, 2) = relative_particle_position[1];
+            A(property_index, 0) = 1;
+            A(property_index, 1) = relative_particle_position[0];
+            A(property_index, 2) = relative_particle_position[1];
             if (dim == 2)
               {
-                A(index, 3) = relative_particle_position[0] * relative_particle_position[1];
+                A(property_index, 3) = relative_particle_position[0] * relative_particle_position[1];
```
Comment (gassmoeller) [inline source/particle/interpolator/bilinear_least_squares.cc]: Shouldnt this line and the three above be `positions_index`?

File: tests/particle_interpolator_bilinear_add_particles.prm [hunk 2]
```diff
@@ -0,0 +1,128 @@
+# A test based on the SolKz benchmark that performs bilinear
```
Comment (gassmoeller) [inline tests/particle_interpolator_bilinear_add_particles.prm]: Please update this test description to explain what this test does now. In particular that it tests to interpolate multiple properties with the bilinear least squares interpolator.

File: source/particle/interpolator/bilinear_least_squares.cc [hunk 3]
```diff
@@ -84,91 +80,106 @@ namespace aspect
...
-        const double threshold = 1e-15;
+        constexpr double threshold = 1e-15;
         unsigned int index_positions = 0;
 
-        // Form the matrix B=A^TA and right hand side A^Tr of the normal equation.
-        A.Tmmult(B, A, false);
-        A.Tvmult(c_ATr,r);
+        // Matrix A can be rank deficient if it does not have full rank, therefore singular.
+        // To circumvent this issue, we solve A^TAx=A^Tr by using singular value
+        // decomposition (SVD).
```
Comment (gassmoeller) [inline source/particle/interpolator/bilinear_least_squares.cc]: This comment was replaced on the master branch in #3394 (probably after you created your branch) by:
`// Form the matrix B=A^TA and right hand side A^Tr of the normal equation.` Please use the new version unless there is value in having the extended description (if so, please reword to form a more concise comment).

**Comment by sac-bsa** [comment]
@gassmoeller I believe that I have resolved all of your concerns with the pull request. Is there anything else that should be done before merging?

**Comment by gassmoeller** [comment]
No, looks good. Thanks!

**Pull request merged**